docs/sdg/knowledge-doc-ingestion.md

```diff
@@ -0,0 +1,96 @@
+# Knowledge Document Ingestion Pipeline Design Proposal
+
+**Author**: Aakanksha Duggal
+
+## 1. Introduction
+
+As part of extending InstructLab's capabilities, this pipeline is designed to support ingestion and processing of various document formats such as Markdown, PDF, DOCX, and more. The goal is to create a unified, modular system that seamlessly integrates with **Synthetic Data Generation (SDG)** and **Retrieval-Augmented Generation (RAG)** workflows, simplifying the process for users while maintaining flexibility for future enhancements.
+
+## 2. Use Case
+
+To enable the ingestion and processing of a wide range of document types, this pipeline must handle formats including:
+
+- Markdown (MD)
+- PDF
+- TXT
+- ASCII Docs
+- DOCX
+- HTML
+- PPTX
+
+This proposal outlines how to build a pluggable system that accommodates these formats and integrates effectively into existing **SDG** and **RAG** workflows.
+
+## 3. Proposed Approach
+
+### 3.1 Custom InstructLab Schema Design
```

Comment:
It is worth pointing out we have an existing `schema` package maintained by the @instructlab/schema-maintainers: https://github.com/instructlab/schema

Right now this is only for Taxonomy schema, but we could extend this with Classes designed specifically for this usecase